intermediate_source/fx_conv_bn_fuser.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+"""
+(beta) Building a Conv/BN fuser in FX
+*******************************************************
+**Author**: `Horace He <https://github.com/chillee>`_
+
+In this tutorial, we are going to use FX to do the following:
+
+1) Find patterns of conv/batch norm.
+2) Fold the batch norm statistics into the convolution weights.
+
+Despite this being a fairly trivial graph rewrite, this has been surprisingly
+difficult to do in PyTorch for quite some time.
+
+We will be building the fuser that exists here:
+https://github.com/pytorch/pytorch/blob/master/torch/fx/experimental/fuser.py
+
+"""
+
+
+######################################################################
+# First, let's get some imports out of the way (we will be using all
+# of these later in the code).
+
+from typing import Type, Dict, Any, Tuple, Iterable
+import copy
+import torch.fx as fx
+import torch
+import torch.nn as nn
+
+######################################################################
+# For this tutorial, we are going to create a model consisting of convolutions
+# and batch norms. Note that this model has some tricky components - some of
+# the modules are hidden within sequentials and one of the modules is wrapped
+# inside of another PyTorch module.
+
+class Wrapper(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.mod = nn.BatchNorm2d(1)
+    def forward(self, x):
+        return self.mod(x)
+
+class M(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = nn.Conv2d(1, 1, 1)
+        self.bn1 = nn.BatchNorm2d(1)
+        self.conv2 = nn.Conv2d(1, 1, 1)
+        self.nested = nn.Sequential(
+            nn.BatchNorm2d(1),
+            nn.Conv2d(1, 1, 1),
+        )
+        self.wrapped = Wrapper()
+
+    def forward(self, x):
+        x = self.conv1(x)
+        x = self.bn1(x)
+        x = self.conv2(x)
+        x = self.nested(x)
+        x = self.wrapped(x)
+        return x
+
+model = M()
+
+model.eval()
+
+######################################################################
+# Fusing Convolution with Batch Norm
+# -----------------------------------------
+# One of the primary challenges with trying to automatically fuse convolution
+# and batch norm in PyTorch is that PyTorch does not provide an easy way of
+# accessing the computational graph. FX resolves that problem.
+
+traced_model = torch.fx.symbolic_trace(model)
+print(traced_model.graph)
+
+######################################################################
+# This gives us a graph representation of our model. Note that both the modules
+# hidden within the sequential as well as the wrapped modue have been inlined
+# into the graph. More information can be found at the FX documentation
+# https://pytorch.org/docs/master/fx.html.
```

Comment:
The docs are pretty long, probably better to point out what specific information you're referring to, e.g. link to the symbolic tracing section and mention this is the default level of abstraction that's traced